ui/app/components/job-page/parts/task-groups.js

```diff
@@ -19,7 +19,7 @@ export default class TaskGroups extends Component.extend(Sortable) {
 
   @action
   gotoTaskGroup(taskGroup) {
-    this.router.transitionTo('jobs.job.task-group', this.job, taskGroup);
+    this.router.transitionTo('jobs.job.task-group', this.job, taskGroup.name);
```

Comment:
`thought` `non-blocking`: We want to be careful here with the `transitionTo` API. It accepts a model object and an identifier. This block of code will be harder to read later on.

Something I'm really digging is the seeker/stumbler paradigm ([page 191](https://abseil.io/resources/swe_at_google.2.pdf)). Maybe we should add a line of code about why `name` can substitute the `taskGroup.id`.

Reply:
Oh boy! Why is `taskGroup` a fragment? I'm trying to find our documentation regarding `taskGroups` in the API, but can't find it.

Either way, great thing to note and point out.

Reply:
Because task groups don't exist as an entity of their own, they are part of the jobspec. When a job runs the task groups get materialized into allocations, but the task groups themselves are just part of the job specification.